app/controllers/api/v1/helpers/test_helper.rb

```diff
@@ -0,0 +1,112 @@
+        amount = subject.question_amount
+        ActiveRecord::Base.transaction do
+          test.questions << subject.questions.sample(amount).sort
+        end
+      rescue ActiveRecord::Rollback
+        error!("Question update failed", 500)
+      end
+
+      def inform_error_for_test_create subject
+        error!("Subject not exists", :forbidden) unless subject
+
+        if @current_user.is_supervisor?
+          error!("Must be a normal user", :forbidden)
+        end
+
+        return unless subject.questions.count < subject.question_amount
+
+        error!("Not enough questions", :forbidden)
+      end
+    end
+
+    helpers do
+      def enqueue_job
+        time = (@test.subject.test_duration + 0.1).minutes
+        CalculateScoreOvertimeJob.set(wait: time)
+                                 .perform_later(@test.id)
+      end
+
+      def submit_test corrected
+        ActiveRecord::Base.transaction do
+          update_test(calculate_score(@answers, corrected))
+          unless corrected.empty?
+            TestQuestion.by_array_of_ids(corrected).update(correct: true)
+          end
+        end
+      end
+
+      def update_saved_test
+        @test.update(pause_time: Time.zone.now)
+      end
+    end
+
+    helpers do
+      def save_answers detail_answers
+        DetailAnswer.by_test_question_ids(@test_questions_ids).delete_all
+        @answers.each_with_index do |answer, index|
+          next if check_invalid_answer answer
+
+          add_answer_to_question(answer, detail_answers, index)
+        end
+        if detail_answers.empty?
+          update_saved_test
+        else
+          create_detail_answers detail_answers
+        end
+      end
+
+      def create_detail_answers detail_answers
+        ActiveRecord::Base.transaction do
+          create_relations detail_answers
+          unless Settings.update_commit.include? params[:commit]
+            update_saved_test
+          end
+        end
+      end
+
+      def update_test score
+        rate = score * 1.0 / @test.subject.question_amount * 100
+        check_pass = rate > @test.subject.pass_score ? 1 : 2
+        @test.update!(status: check_pass, score: score, end_time: Time.zone.now)
+      end
+    end
+
+    helpers do
+      def post_data_handle
+        @test_questions_ids = params.dig(:test, :test_question)&.keys
+        data = params.dig(:test, :test_question)&.values
+        @answers = data.map{|i| i[:first_answer_id] || i[:answer_ids]} if data
+
+        return if @answers && @test_questions_ids
+
+        error!("data not valid", 422)
+      end
+
+      def true_answers
+        @true_answers = @test.questions.includes(:answers)
+                             .where(answers: {is_correct: true})
+                             .map{|i| i.answers.map(&:id)}
+      end
+
+      def check_invalid_answer answer
+        return true if answer == "" ||
+                       (answer.is_a?(Array) && answer.size == 1)
+
+        false
+      end
```

Comment:
```suggestion
      def check_invalid_answer answer
        answer == "" || (answer.is_a?(Array) && answer.size == 1)
      end
```
ri dc hok 
chị làm theo quy tắt kết quả của method là kết quả của dòng lệnh sau cùng dc run nè
ko cần return đồ cho màu mè =))